### PR TITLE
Change dnc extension mode

### PR DESCRIPTION
**Pull request by shriroopjoshi**
Change dnc extension mode from experimental to preview

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
dnc

File: src/dnc/azext_dnc/azext_metadata.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-    "azext.isExperimental": true,
+    "azext.isExperimental": false,
```
Comment (kairu-ms) [inline src/dnc/azext_dnc/azext_metadata.json:2]:
```suggestion
    "azext.isPreview": true,
```

File: src/dnc/azext_dnc/generated/commands.py [hunk 2]
```diff
@@ -60,5 +60,5 @@ def load_command_table(self, _):
         g.custom_command('delete', 'dnc_delegated_subnet_service_delete', supports_no_wait=True, confirmation=True)
         g.custom_wait_command('wait', 'dnc_delegated_subnet_service_show')
 
-    with self.command_group('dnc', is_experimental=True):
+    with self.command_group('dnc', is_experimental=False):
```
Comment (kairu-ms) [inline src/dnc/azext_dnc/generated/commands.py:63]:
```suggestion
    with self.command_group('dnc', is_preview=True):
```

**Review by kairu-ms — COMMENTED** [review]
Please update extension version in setup.py and history note in HISTORY.rst

File: src/dnc/azext_dnc/azext_metadata.json (hunk 1 above)
Comment (shriroopjoshi) [reply src/dnc/azext_dnc/azext_metadata.json]: Thanks, updated

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**